src/inmanta/data/model.py

```diff
@@ -747,31 +748,67 @@ class LoginReturn(BaseModel):
     user: User
 
 
+def is_resource_id(v: ResourceIdStr) -> str:
+    # For validation, we rely on the parse_id method raising a ValueError for invalid resource ids.
+    return resources.Id.parse_id(v).resource_str()
```

Comment:
Mypy was complaining when typing this as `def is_resource_id(v: str) -> ResourceIdStr:` 
```
src/inmanta/data/model.py:753: error: Argument 1 to "parse_id" of "Id" has incompatible type "str"; expected "ResourceVersionIdStr | ResourceIdStr"  [arg-type]
```

Added a cast to `ResourceIdStr` in the call to `parse_id`

Reply:
Ah. I see. That is unexpected to me. I would expect the `Id.parse_id()` method to accept a string as well, instead of only `ResourceVersionIdStr | ResourceIdStr`. I would do the following:
* Remove the type cast, because at the moment the type cast is actually a lie. We are not certain the string is actually a ResourceIdStr.
* I would ask on Slack whether there is a reason to not accept the string type here. The change was made in https://github.com/inmanta/inmanta-core/commit/85aa5e4b629d287b1e2c5ca625843192cd3b00f5

Reply:
[Slack discussion](https://inmanta.slack.com/archives/CKRF0C8R3/p1724827928487769) for reference